### PR TITLE
Keep index and time tags for single column

### LINKED ISSUE #867: Keep index and time index tags when selecting a single column
Currently, we remove `index` and `time_index` tags from a Woodwork Column if it is created off of a Table via `loc`, `iloc`, or `__getitem__`. We should keep the index tags so that `df.ww.columns['col_name'] == df.ww['col_name'].ww.schema`

### PR DESCRIPTION
**Pull request by jeff-hernandez**
Closes #867

### CODE REVIEW

File: woodwork/indexers.py [hunk 1]
```diff
@@ -42,7 +42,6 @@ def _process_selection(selection, original_data):
         elif _is_dataframe(original_data):
             # Selecting a single column from a DataFrame
             schema = original_data.ww.schema.columns[selection.name]
-            schema.semantic_tags = schema.semantic_tags - {'index'} - {'time_index'}
             if schema.use_standard_tags:
                 schema.semantic_tags |= schema.logical_type.standard_tags
```
Comment (tamargrey) [inline woodwork/indexers.py:46]: Now that we're keeping the `index` tag, I'm not sure that we want to add standard tags back like this (and at the table accessor getitem) since at the Woodwork table level, you either have standard tags or the index tag but not both. 

Not sure if maybe at the column level it does make sense to have both. @thehomebrewnerd thoughts?

Comment (thehomebrewnerd) [reply woodwork/indexers.py:46]: Yeah, my thought going into this was that when we grab a column from the dataframe, the column schema for the individual column should match the schema that is in the dataframe for that column - that way no mater where the user looks they see the same info.

tldr: I vote for not adding back in standard tags.

File: docs/source/release_notes.rst [hunk 2]
```diff
@@ -12,6 +12,18 @@ Release Notes
 
 .. Thanks to the following people for contributing to this release:
 
+Future Release
+==============
+    * Enhancements
+    * Fixes
+    * Changes
+        * Keep index and time index tags for single column (:pr:`888`)
```
Comment (thehomebrewnerd) [inline docs/source/release_notes.rst:20]: I'd suggest a little more context here - like adding `...when selecting from a table`

File: woodwork/indexers.py (hunk 1 above)
Comment (jeff-hernandez) [reply woodwork/indexers.py]: ✅   ceccc4d

File: docs/source/release_notes.rst (hunk 2 above)
Comment (jeff-hernandez) [reply docs/source/release_notes.rst]: ✅  fe673e2

File: woodwork/table_accessor.py [hunk 3]
```diff
@@ -206,7 +206,6 @@ def __getitem__(self, key):
 
         series = self._dataframe[key]
         column = copy.deepcopy(self._schema.columns[key])
-        column.semantic_tags -= {'index', 'time_index'}
         if column.use_standard_tags:
             column.semantic_tags |= column.logical_type.standard_tags
```
Comment (jeff-hernandez) [inline woodwork/table_accessor.py:210]: Do we want to remove standard tags here too?

Comment (tamargrey) [reply woodwork/table_accessor.py:210]: yes! It should have the same behavior as the indexers

Comment (jeff-hernandez) [reply woodwork/table_accessor.py]: ✅  5fe0b07

**Review by thehomebrewnerd — APPROVED** [review]
LGTM

**Review by tamargrey — APPROVED** [review]
